internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,335 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package fuzzbinder
+
+// This file provides the scraper module that helps to get the fuzzing statistics
+// from the ClusterFuzz and OSS-Fuzz reports.
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"strconv"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// CoverageBucket is the OSS-Fuzz Google Cloud Storage bucket containing
+// the coverage reports.
+const CoverageBucket = "oss-fuzz-coverage"
+
+// OSS-Fuzz uses Clang to generate coverage summary reports.The coverage
+// summary contains fine-grained data related to coverage metrics.
+// CoverageSummary contains the coverage report summary.
+type CoverageSummary struct {
+	Data    []CoverageSummaryData `json:"data"`
+	Type    interface{}           `json:"type"`
+	Version interface{}           `json:"version"`
+}
+
+// CoverageSummaryData contains the data of the coverage report summary.
+type CoverageSummaryData struct {
+	Files  interface{}                     `json:"files"`
+	Totals map[string](map[string]float64) `json:"totals"`
+}
+
+// getBucket gets a Google Cloud Storage bucket given its name.
+func getBucket(bucketName string) (*storage.BucketHandle, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	return bucket, nil
+}
+
+// listBlobs gets all the objects in a Google Cloud Storage bucket
+// under a given relative path.
+func listBlobs(bucket *storage.BucketHandle, relativePath string) ([]string, error) {
+	var blobs []string
+	ctx := context.Background()
+	query := &storage.Query{Prefix: relativePath}
+	objects := bucket.Objects(ctx, query)
+	for {
+		attrs, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket.Objects: %v", err)
+		}
+		blobs = append(blobs, attrs.Name)
+	}
+	return blobs, nil
+}
+
+// getBlob gets the file reader of a blob in a Google Cloud Storage bucket.
+func getBlob(bucket *storage.BucketHandle, blobName string) (*storage.Reader, error) {
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", blobName, err)
```

Comment:
nit: to be consistent with error messages throughout the code repo.

```suggestion
		return nil, fmt.Errorf("could not create a new reader for blob %q: %v", blobName, err)
```